### ISSUE TITLE
crash on csv export

### ISSUE THREAD

**Issue by timabell** [wbaumann/SmartReceiptsLibrary#10]
Hi, great app.

It reliably crashes when I choose CSV export. I deleted the `Android/data/wb.receiptspro` folder and started again in case it was bad data but it's still doing it.

The only message is "Unfortunately, Smart Receipts Pro has stopped.  [OK]"

I'm running v3.3.3 via F-droid on cyanogen-mod (12-29150322-NIGHTLY-mako) on a nexus 4.

I'm happy to collect more info if you can point me in the right direction.

This is the only thing that looked interesting to me when I pointed `adb logcat` at my phone and recreated the crash:

	E/WindowManager(25782): android.view.WindowLeaked: Activity wb.receiptspro.SmartReceiptsProActivity has leaked window com.android.internal.policy.impl.PhoneWindow$DecorView{22daef6f V.E..... R......D 0,0-729,192} that was orig
	inally added here
	E/WindowManager(25782):         at android.view.ViewRootImpl.<init>(ViewRootImpl.java:363)
	E/WindowManager(25782):         at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:261)
	E/WindowManager(25782):         at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:69)
	E/WindowManager(25782):         at android.app.Dialog.show(Dialog.java:298)
	E/WindowManager(25782):         at android.app.ProgressDialog.show(ProgressDialog.java:116)
	E/WindowManager(25782):         at android.app.ProgressDialog.show(ProgressDialog.java:104)
	E/WindowManager(25782):         at co.smartreceipts.android.workers.EmailAssistant$2.onClick(EmailAssistant.java:181)
	E/WindowManager(25782):         at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:160)
	E/WindowManager(25782):         at android.os.Handler.dispatchMessage(Handler.java:102)
	E/WindowManager(25782):         at android.os.Looper.loop(Looper.java:135)
	E/WindowManager(25782):         at android.app.ActivityThread.main(ActivityThread.java:5254)
	E/WindowManager(25782):         at java.lang.reflect.Method.invoke(Native Method)
	E/WindowManager(25782):         at java.lang.reflect.Method.invoke(Method.java:372)
	E/WindowManager(25782):         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
	E/WindowManager(25782):         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)

I'm no android dev so it doesn't mean much to me.

**Comment by timabell** [comment]
Downgrading to 2.2.1 fixes it again but loses all the entries I'd made. There are no versions between these two available on f-droid.

https://f-droid.org/repository/browse/?fdfilter=smart+receipts&fdid=wb.receiptspro

**Comment by wbaumann** [comment]
Hi Tim,

Sorry about this and I'll get it fixed up asap. The logcat trace you sent doesn't seem to contain a NullPointerException, but someone sent me a support email the other day asking about a similar issue:

```
E/AndroidRuntime(29257): java.lang.RuntimeException: An error occured while executing doInBackground()
E/AndroidRuntime(29257): 	at android.os.AsyncTask$3.done(AsyncTask.java:300)
E/AndroidRuntime(29257): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
E/AndroidRuntime(29257): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
E/AndroidRuntime(29257): 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
E/AndroidRuntime(29257): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
E/AndroidRuntime(29257): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime(29257): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime(29257): 	at java.lang.Thread.run(Thread.java:841)
E/AndroidRuntime(29257): Caused by: java.lang.NullPointerException
E/AndroidRuntime(29257): 	at co.smartreceipts.android.model.impl.ImmutableNetPriceImpl.getCurrencyCode(ImmutableNetPriceImpl.java:112)
E/AndroidRuntime(29257): 	at co.smartreceipts.android.workers.reports.columns.DistanceTableColumns.getFooterValueAt(DistanceTableColumns.java:117)
E/AndroidRuntime(29257): 	at co.smartreceipts.android.workers.reports.columns.DistanceTableColumns.getValueAt(DistanceTableColumns.java:54)
E/AndroidRuntime(29257): 	at co.smartreceipts.android.workers.reports.writers.CsvTableGenerator.write(CsvTableGenerator.java:28)
E/AndroidRuntime(29257): 	at co.smartreceipts.android.workers.EmailAssistant$EmailAttachmentWriter.doInBackground(EmailAssistant.java:585)
E/AndroidRuntime(29257): 	at co.smartreceipts.android.workers.EmailAssistant$EmailAttachmentWriter.doInBackground(EmailAssistant.java:339)
E/AndroidRuntime(29257): 	at android.os.AsyncTask$2.call(AsyncTask.java:288)
E/AndroidRuntime(29257): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/AndroidRuntime(29257): 	... 4 more```

Would you be able to check through your logcat logs and let me know if you see a similar stack? It should be the Error (E) logs that appear immediately preceding `NullPointerException`.

Thanks so much for bringing this to my attention! 

Best,
Will

**Comment by wbaumann** [comment]
Should be resolved with the 3.3.4 release
